Me: Accessibility: Allow tabbing through Interface Language options using keyboard

ramonjd opened the pull request:
This PR address #21222 by adding interactive/focusable capabilities to the list of languages in the language picker.

![jan-24-2018 13-11-48](https://user-images.githubusercontent.com/6458278/35311622-8b1b94a6-010b-11e8-9ad6-a3cfb44b6894.gif)

## Testing

1. Open the language picker and tab through the languages

**Expectation**
You should be able to tab through the language items, and select them by hitting the 'space' or 'enter' keys.

ramonjd commented:
hi @ryelle this one also okay to merge up in your opinion? 
thanks

ryelle commented:
Can you rebase this on master, now that the modal component has been updated? That's preventing me from testing this. I'm also having trouble opening the modal itself, since the language picker "button" isn't really a button -- I can't easily get to it via keyboard navigation.

Code-wise this looks OK, and having it be accessible at all is a great step.

If you want to/have time to improve this further, I'd suggest letting the screen reader "see" the native language name (by removing the aria-label), paired with the `lang` attribute to indicate the pronunciation. [This post has some basic information on this](https://developer.paciellogroup.com/blog/2016/06/using-the-html-lang-attribute/), but just dropping the lang on the div doesn't seem to work. I think there's an issue with VoiceOver + lang on buttons. I found [this post on accessible language pickers](http://terrillthompson.com/tests/lang-picker.html), where it uses links, and that seems to work (despite not being the "correct" element, I think the fact that it reads in the correct language outweighs that).

ramonjd commented:
Thanks @ryelle 

I've added a keyboard event to the Language Picker so it should be possible to open it. I've also removed the `aria-label` and added the `lang` attribute, but after reading the accessible lang picker article it might need some more thought in general. Perhaps a list of label/radio inputs might be more appropriate. 

This exercise made me realize there the 'Primary site' drop down is also inaccessible via keyboard, and that there's a lot more to do, at least on this page. It's a good candidate for another PR, e.g., to create an 'accessible' higher order container component that adds `tabIndex` and keyboard handler props.

ryelle commented:
> I've added a keyboard event to the Language Picker so it should be possible to open it.

Great :) I noticed there was no focus style, so I've added that in 0180c13.

> Perhaps a list of label/radio inputs might be more appropriate.

I agree. Unfortunately, it seems that VoiceOver ignores the `lang` attribute on input controls (including things with `role=button`), and falls back to the OS language. I'm not sure that's a problem we need to solve immediately (users are probably trying to switch wp.com into the language they use on the OS), but it is bothering me during my testing 😅 (and it's an issue on that testing page, too).

> This exercise made me realize there the 'Primary site' drop down is also inaccessible via keyboard, and that there's a lot more to do, at least on this page.

Yeah, these older pages are all good candidates for accessibility iteration 🙂 

The keyboard helper idea has come up before, and [we actually do something like that in the woocommerce extension](https://github.com/Automattic/wp-calypso/tree/master/client/extensions/woocommerce/lib/get-keyboard-handler), but a better approach would be to switch to semantic elements (buttons, ideally, links when appropriate) to get the right functionality across assistive tech.

…but maybe this is needed for faster fixes, where we can't make those changes immediately. 🤔
